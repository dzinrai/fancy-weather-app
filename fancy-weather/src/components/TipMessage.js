import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useTranslation } from 'react-i18next';

function TipMessage(props) {
    const { t } = useTranslation();
    const messageExist = props.message;

	if (messageExist && messageExist.listening) {
        const message = [];
        message.push(`"${t('Weather')}" - ${t('Listen audio weather')}\n`);
        message.push(`"${t('Stop')}" - ${t('Stop audio weather')}\n`);
        message.push(`"${t('Language')}" - ${t('Change language')}\n`);
        message.push(`"${t('Off')}" - ${t('Turn microphone off')}\n`);
		return (
			<div className="message__container error__log">
				<div className={"error-span"}>
                    {message.map((line, i) => 
                        <span key={i+1}>{line}</span>
                    )}
					<FontAwesomeIcon icon='times' onClick={props.clearMessage} className='close__btn' />
				</div>
			</div>
		);
	} else {
		return null;
	}

}

export default TipMessage;