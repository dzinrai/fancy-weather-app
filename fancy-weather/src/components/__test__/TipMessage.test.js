import React from 'react';
import {cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import TipMessage from '../TipMessage';

afterEach(cleanup);

test('proper message', () => {
    const tree = renderer.create(<TipMessage message={null} />).toJSON();
    expect(tree).toMatchSnapshot();
})

test('proper message', () => {
    const tree = renderer.create(<TipMessage message={{listening: false}} />).toJSON();
    expect(tree).toMatchSnapshot();
})

test('proper message', () => {
    const tree = renderer.create(<TipMessage message={{listening: true}} />).toJSON();
    expect(tree).toMatchSnapshot();
})