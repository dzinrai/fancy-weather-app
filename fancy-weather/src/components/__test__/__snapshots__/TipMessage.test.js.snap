// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proper message 1`] = `null`;

exports[`proper message 2`] = `null`;

exports[`proper message 3`] = `
<div
  className="message__container error__log"
>
  <div
    className="error-span"
  >
    <span>
      "Weather" - Listen audio weather

    </span>
    <span>
      "Stop" - Stop audio weather

    </span>
    <span>
      "Language" - Change language

    </span>
    <span>
      "Off" - Turn microphone off

    </span>
  </div>
</div>
`;
